File: rb/features.rb
# WorldBankData SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module WorldBankDataFeatures
  def self.make_feature(name)
    case name
    when "base"
      WorldBankDataBaseFeature.new
    when "ratelimit"
      WorldBankDataRatelimitFeature.new
    when "retry"
      WorldBankDataRetryFeature.new
    when "test"
      WorldBankDataTestFeature.new
    when "timeout"
      WorldBankDataTimeoutFeature.new
    else
      WorldBankDataBaseFeature.new
    end
  end
end

File: py/features.py
# WorldBankData SDK feature factory

from feature.base_feature import WorldBankDataBaseFeature
from feature.test_feature import WorldBankDataTestFeature


def _make_feature(name):
    features = {
        "base": lambda: WorldBankDataBaseFeature(),
        "test": lambda: WorldBankDataTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
